File: python/leetcode013.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        table = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        res = 0
        ls = len(s)
        for i in range(ls):

            if i > 0  and table[s[i]] > table[s[i-1]]:
                res = res + table[s[i]] - 2*table[s[i-1]]
            else:
                res = res + table[s[i]]
        return res

if __name__ == "__main__":
    s = Solution()
    print(s.romanToInt('IV'))

File: leetcodec++/leetcode013.cpp
//
// Created by 黄敏辉 on 2019-04-25.
//

using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        unordered_map <char, int> hash = {{'I',1},{'V',5},{'X',10},{'L',50},{'C',100},{'D',500}, {'M',1000}};
        int res = 0;
        int i = 0;
        cout << s.length()<<endl;
        while(i < s.length()){
            if (i > 0 && hash[s[i]] > hash[s[i-1]]){
                res = res + hash[s[i]] - 2*hash[s[i-1]];
            }else
                res = res + hash[s[i]];
            cout << res <<endl;
            i++;
        }
        return res;
    }
};